records_mover/records/hints.py

```diff
@@ -201,3 +204,178 @@ def sniff_hints(fileobj: IO[bytes],
+
+# The trick here works on Literal[True, False] but not on bool:
+#
+# https://github.com/python/mypy/issues/6366#issuecomment-560369716
+HintHeaderRow = Literal[True, False]
+
+HintDoublequote = Literal[True, False]
+
+
+# TODO: This None is a bug in the spec, right?
+HintDateFormat = Literal[None, 'YYYY-MM-DD', 'MM-DD-YYYY', 'DD-MM-YYYY', 'MM/DD/YY']
+
+HintTimeOnlyFormat = Literal["HH12:MI AM", "HH24:MI:SS"]
+
+HintDateTimeFormatTz = Literal["YYYY-MM-DD HH:MI:SSOF",
+                               "YYYY-MM-DD HH:MI:SS",
+                               "YYYY-MM-DD HH24:MI:SSOF", # TODO: this is listed twice - bug in spec?
+                               "YYYY-MM-DD HH24:MI:SSOF",
+                               "MM/DD/YY HH24:MI"]
+
+HintDateTimeFormat = Literal["YYYY-MM-DD HH24:MI:SS",
+                             'YYYY-MM-DD HH:MI:SS', # TODO this isn't in spec valid, but is part of a variant
+                             "YYYY-MM-DD HH12:MI AM",
+                             "MM/DD/YY HH24:MI"]
+
+
+HintFieldDelimiter = str
+
+HintRecordTerminator = str
+
+HintQuoteChar = str
+
+
+BootstrappingRecordsHints = TypedDict('BootstrappingRecordsHints',
+                                      {
+                                          'quoting': HintQuoting,
+                                          'header-row': HintHeaderRow,
+                                          'field-delimiter': HintFieldDelimiter,
+                                          'encoding': HintEncoding,
+                                          'escape': HintEscape,
+                                          'compression': HintCompression,
+                                      },
+                                      total=False)
+
+
+# TODO: these should live in hints.py - existing stuff should probably move
+HintName = Literal["header-row",
+                   "field-delimiter",
+                   "compression",
+                   "record-terminator",
+                   "quoting",
+                   "quotechar",
+                   "doublequote",
+                   "escape",
+                   "encoding",
+                   "dateformat",
+                   "timeonlyformat",
+                   "datetimeformattz",
+                   "datetimeformat"]
+
+HintT = TypeVar('HintT')
+
+
+class Hint(Generic[HintT], metaclass=ABCMeta):
+    ...
+    @abstractmethod
+    def validate(self,
+                 hints: RecordsHints,
+                 fail_if_cant_handle_hint: bool) -> HintT:
+        ...
+
+
+class StringHint(Hint[str]):
+    def __init__(self,
+                 hint_name: HintName,
+                 default: str) -> None:
+        self.default = default
+        self.hint_name = hint_name
+
+    def validate(self,
+                 hints: RecordsHints,
+                 fail_if_cant_handle_hint: bool) -> str:
+        x: object = hints[self.hint_name]
+        if isinstance(x, str):
+            return x
+        else:
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker